### PR TITLE
fix/sg: fix mangled log output from `sg start` and `sg run`

### PR DESCRIPTION
**Pull request by bobheadxi**
Right now `sg run` / `sg start` can horribly mangle multi-line output. A nicely annotated report from @unknwon:

image

Replacing the "buffered process logger" thing with https://github.com/bobheadxi/streamline which powers `sourcegraph/run` etc (fairly reliably if I do say so myself) fixes this for a few cases where I can reliably repro wonky misordered output 😁 

## Test plan

`sg start dotcom` with `sg.config.overwrite.yaml`:

```yaml
commands:
  enterprise-portal:
    env:
      SRC_LOG_LEVEL: debug
      PG_QUERY_LOGGING: true
```

Log scope `pgx.devtracer` is consistently formatted ✅ , even with high volume of logs

image

Also don't see anything suspicious happening after running for a while

### CODE REVIEW

File: dev/sg/internal/run/logger.go [hunk 1]
```diff
@@ -31,20 +34,29 @@ var (
...
-			out.Writef(lineFormat, output.StyleBold, color, name, output.StyleReset, data)
-		}()
-	}
+	w, stream := pipe.NewStream()
+	go func() {
+		var mux sync.Mutex
+		<-start
+		err := stream.Stream(func(line string) {
+			mux.Lock()
+			out.Writef(lineFormat, output.StyleBold, color, name, output.StyleReset, line)
```
Comment (bobheadxi) [inline dev/sg/internal/run/logger.go:49]: `*output.Output` is not concurrency-safe

**Comment by unknwon** [comment]
Checked out the branch locally, seems still happening? 🤔 

CleanShot 2024-06-20 at 18 59 18@2x

**Comment by unknwon** [comment]
CleanShot 2024-06-20 at 19 01 12@2x

Seems fixed it!

**Review by unknwon — APPROVED** [review]

**Comment by jhchabran** [comment]
Thanks!

**Pull request merged**